src/libcollections/string.rs

```diff
@@ -1881,6 +1881,41 @@ impl Into<Vec<u8>> for String {
     }
 }
 
+#[stable(feature = "stringfromchars", since = "1.12.0")]
+impl<'a> From<&'a [char]> for String {
+    #[inline]
+    fn from(v: &'a [char]) -> String {
+        let mut s = String::with_capacity(v.len());
```

Comment:
What do you think about the suggestion from https://github.com/rust-lang/rust/pull/35054#issuecomment-235428354?
`char` usually implies that we work with human language so we can use domain knowledge.
`len + len / 8`(`=1.125`) or `len + len / 16`(`=1.0625`) cover most of European languages.
For reference,
`ratio(en) ≈ 1.002`
`ratio(fr) ≈ 1.040`
`ratio(de) ≈ 1.016`
`ratio(hu) ≈ 1.091` (lots of diacritics)